test/integration/operator_run_bundle_test.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modified from https://github.com/kubernetes-sigs/kubebuilder/tree/39224f0/test/e2e/v3
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-sdk/internal/olm/operator"
+	"github.com/operator-framework/operator-sdk/internal/olm/operator/bundle"
+	testutils "github.com/operator-framework/operator-sdk/test/internal"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	wait "k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kbtestutils "sigs.k8s.io/kubebuilder/test/e2e/utils"
+)
+
+// TODO: After run bundle is complete, remove the boolean variable and the if statements.
+var runBundleImplemented = false
+
+var _ = Describe("run bundle", func() {
+	var (
+		err error
+		tc  testutils.TestContext
+	)
+	BeforeEach(func() {
+		By("creating a new context")
+		// For this test suite, the bundle image name is equal to tc.ImageName
+		tc, err = testutils.NewTestContext("GO111MODULE=on")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(tc.Prepare()).To(Succeed())
+
+		createOperator(tc, defaultOperatorName)
+	})
+
+	AfterEach(func() {
+		By("removing container image and working dir")
+		tc.Destroy()
+	})
+
+	It("Run Bundle Basic", func() {
+		if !runBundleImplemented {
+			Skip("Run bundle not yet implemented")
+		}
+
+		cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+		err := cfg.Load()
+		Expect(err).NotTo(HaveOccurred())
+
+		i := bundle.NewInstall(cfg)
+		i.BundleImage = tc.ImageName
```

Comment:
Since it is doing a e2e test then we should use the makefile targets instead. See; 

https://github.com/operator-framework/operator-sdk/blob/master/test/e2e-helm/e2e_helm_olm_test.go#L62-L63

Reply:
The bundle image is an argument to the run bundle command. When the command is passed, the `Install` struct is populated. It is referenced here: https://github.com/operator-framework/operator-sdk/blob/master/internal/olm/operator/bundle/install.go#L42

Reply:
why not we do not use operator-sdk run bundle here? IMO : It is an e2e test so we should use the CLI commands otherwise we could cover these scenarios with unit tests.